packages/testcontainers/src/wait-strategies/http-wait-strategy.ts

```diff
@@ -66,9 +66,10 @@ export class HttpWaitStrategy extends AbstractWaitStrategy {
 
   public async waitUntilReady(container: Dockerode.Container, boundPorts: BoundPorts): Promise<void> {
     log.debug(`Waiting for HTTP...`, { containerId: container.id });
-    const client = await getContainerRuntimeClient();
 
-    await new IntervalRetry<Response | undefined, Error>(this.readTimeout).retryUntil(
+    const waitingFinished = { value: false };
```

Comment:
Added fix after your comment for stopping second waiter when one of them stopped.